### DISCUSSION TITLE
Tuomas Artman - Linear sync engine

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#178] [Announcements]
- https://twitter.com/artman?lang=en

### Background

- [x] We chatted with Jori back in episode 44 about linear but for our new listeners could you give them a refresher on what linear is?

### Sync Engine

- [x] Linear launched with syncing as a first class feature. What was the justification for taking on that level of complexity at the beginning of the product?
- [x] Is linear your first time building a real time sync engine? Seems complicated!
- [x] Did you use an off the shelf sync library or did you write your own solution?
- [x] Are you using CRDTs?
- [x] In basic terms how did the first engine work? What are the benefits
- [x] Are there any unique features of linear system compared to other systems?
- [x] How much of the data in the app is handled by the real time sync engine?
- [x] What were the goals of the new engine?
- [x] What's the biggest challenge you've had to solve for scaling real time sync?
- [x] How did you fix bootstrap speed? (lazy) Did it effect DX at all?
  - [x] How did you fix your graphql probelm?
  - [x] How did you make the database faster?
- [x] How do you handle updating the UI? What does DX look like on the front end?

### Future

- [ ] Do you think all apps should be offline first?

**Comment by hipstersmoothie** [comment]
- https://ui.shadcn.com/
- supabase.com
  - https://socket.dev/npm/package/y-supabase

**Comment by just-be-dev** [comment]
- https://hurl.dev/
- https://fly.io/blog/litefs-cloud/
